test/testrunner.py

```diff
@@ -187,8 +187,15 @@ def run_sagemaker_remote_tests(images, pytest_cache_params):
         if not images:
             return
         pool_number = len(images)
-        with Pool(pool_number) as p:
-            p.starmap(sm_utils.execute_sagemaker_remote_tests, [[image, pytest_cache_params] for image in images])
+        # Using Manager().dict() since it's a tread safe dictionary
+        global_pytest_cache = Manager().dict()
+        try:
+            with Pool(pool_number) as p:
+                p.starmap(sm_utils.execute_sagemaker_remote_tests,
+                                  [[i, images[i], global_pytest_cache, pytest_cache_params] for i in range(0, pool_number)])
```

Comment:
Using `range(0, x)` is equivalent to `range(x)`.

Also please correct the indentation to:
```suggestion
                p.starmap(
                    sm_utils.execute_sagemaker_remote_tests,
                    [[i, images[i], global_pytest_cache, pytest_cache_params] for i in range(pool_number)]
                )
```